week4-ts/src/components/SearchSection.tsx

```diff
@@ -0,0 +1,87 @@
+import styled from "styled-components";
+import { useState, useEffect } from "react";
+
+interface SearchSectionProps {
+  onClick: (isCheck: boolean, myTown: string) => void;
+}
+
+export default function SearchSection({ onClick }: SearchSectionProps) {
+  const [isCheck, setIsCheck] = useState<boolean>(false);
+  const [myTown, setMyTown] = useState<string>("");
+
+  useEffect(() => {
+    if (isCheck) setMyTown("");
+  }, [isCheck]);
+
+  return (
+    <StWrapper>
+      <StCheckBoxWrapper>
+        <label htmlFor="checkbox">지역 기반으로 검색</label>
+        <input
+          type="checkbox"
+          id="checkbox"
+          onChange={() => setIsCheck(!isCheck)}
+        />
+      </StCheckBoxWrapper>
+
+      <StTextInputWrapper>
+        <label htmlFor="text">우리 동네는 여기에요</label>
+        <input
+          type="text"
+          placeholder="지역을 입력해주세요."
```

Comment:
여긴 id 속성이 빠진 것 같네요??

```suggestion
        <label htmlFor="text">우리 동네는 여기에요</label>
        <input
          type="text"
          id="text"
          placeholder="지역을 입력해주세요."
```

Reply:
시력 3.0이세요??